### ISSUE TITLE
Provide a mechanism for longer-lived authentication sessions

### ISSUE THREAD

**Issue by jsimpso** [canonical/jimm#1018]
Hi,

This is a feature request to introduce some mechanism for user configurable long-lived authentication sessions.

I'm part of a team which holds a shared responsibility over hundreds of Juju models. The way we interact with those models is via a bastion server (network access restricted via VPN and firewall, access restricted via LDAP group membership), where each model is only accessible to a single service account (access to which is restricted by sudoers rules applied to subgroups of users who should have access to that model).

For example, I'm able to `sudo -iu my-production-model` to access the local user account, who then has access to a Juju client which is configured to talk to the model.

For the models which we've migrated to an internal JAAS controller, we're experiencing some friction from the fact that we need to re-authenticate via SSO daily, since JIMM's macaroons are only valid for 24 hours.

It would really help to reduce this friction if there was some kind of capability to configure the authorized session time (per-user, per-model, per-controller even). 

I recognise that this may introduce an increased security risk, which is acceptable to us given the security mechanisms we currently have in place, but may not be acceptable to others. If you have any alternate suggestions which could help us to reduce the re-authentication friction while maintaining that security posture, please let me know!

Thanks

**Comment by ale8k** [comment]
Hi @jsimpso , has there been any updates on this?

**Comment by jsimpso** [comment]
Apologies for not coming back to that one. IIRC we discovered shortly after this that JAAS/JIMM were going to undergo a major overhaul, so I think what we have today (or what we're working towards deploying) may be entirely different to what we had when this issue was raised.

I think @alejdg you're probably best positioned to understand if this is still relevant, would you mind weighing in please?

**Comment by alejdg** [comment]
Yeah, this was raised in another time, based on an old JIMM version we had deployed. Since the new versions (machine and k8s) support the macaroon expiry we are good to close this. 

We still have the old jaas-candid deployment as @kian99 pointed-out but there's barely anything connecting to it and in the future everything will use the new deployments.

**Comment by jsimpso** [comment]
The need for this is resolved by service accounts,so closing. Thanks!
